### DISCUSSION TITLE
Possibility of a VectorPostprocessor taking the area average  of a value

### DISCUSSION THREAD

**Discussion by AnthonyB08** [dpploy/engy-5310#44] [Q&A] [answered]
Does anyone know of any MOOSE source file or code implementation that can take the area average of a variable. 

Purpose: 
I want to slice my domain into x number of shells that take the area-average value of u. So that I can plot the value of u against the axial length of my domain.

**Comment by dealmeidavf** [comment]
The Poisson-Dirichlet energy is an [example](https://nbviewer.jupyter.org/github/dpploy/engy-5310/blob/main/notebooks/06-poisson-1d-dirichlet.ipynb#energypostpro) of the area integration of a variable. You just have to divided by the area of the domain.

**Comment by AnthonyB08** [comment, marked as answer]
Cylindrical Average (provided by MOOSE) computes the area average by dividing the domain into 50 shells. 

![image](https://user-images.githubusercontent.com/55299169/118982881-da3b4280-b949-11eb-88ce-b0413184d92e.png)

![image](https://user-images.githubusercontent.com/55299169/118982935-e7583180-b949-11eb-813d-97b7fb80b80b.png)
